scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala

```diff
@@ -31,8 +31,8 @@ trait GrantHandler {
       }.getOrElse {
         handler.createAccessToken(authInfo)
       }
-      case Some(token) => Future.successful(token)
-      case None => handler.createAccessToken(authInfo)
+      case Some(token) if !token.isExpired => Future.successful(token)
```

Comment:
The thing with implicit grant flows is they don't support refresh tokens for a security reason - that you can't trust a web browser with long lived tokens. So refreshing the token for them automatically is essentially doing the same thing as just giving them a refresh token. For implicit grant, there really is no way that I'm aware of besides requiring a user to get a new token by authenticating again every time the token expires, it is the side-effect of choosing to use this grant flow.